### PR TITLE
feat(media): add media directive as $mdMedia wrapper for classes

### PR DESCRIPTION
**Pull request by DevVersion**
This directive allows the user to add specific classes on specific screen-sizes / breakpoints.

**Notes**
- Tests are currently manually (not dynamic using a loop - I can change that if desired)
- mdMedia directive is currently listed under directives / components (Is that the right way?) 

@EladBezalel @ThomasBurleson - Please correct me if the directive should be located where-else (and about the dynamic tests)

PS: Work in progress :+1:

### CODE REVIEW

File: src/components/media/media.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      }
+    }];
+  }
+
+  /**
+   * Converts snake_case to camelCase.
+   * Also there is special case for Moz prefix starting with upper case letter.
+   * @param name Name to normalize
+   */
+  function directiveNormalize(name) {
```
Comment (DevVersion) [inline src/components/media/media.js]: And I suggest this method to be added to `$mdUtil`

**Comment by ThomasBurleson** [comment]
This is the approach we want to pursue with the Layout API. 

PleaseCurrently we support breakpoint triggers for classes using this:

```html
<div ng-class="{$mdMedia('gt-sm'):'blue'}"`> 
    Testing breakpoint changes to trigger class changes
</div>
```

**Comment by DevVersion** [comment]
@ThomasBurleson So whats now with #3815?

**Comment by ThomasBurleson** [comment]
@DevVersion - thanks for the reminder. Let me update that issue. To be honest I had forgotten that this feature already exists. Your PR made me realize the above solution:

http://codepen.io/team/AngularMaterial/pen/YweVWZ

```html
<md-card ng-repeat="it in products.catalog" flex="22" flex-sm="46" flex-xs="80" ng-class="{shaded:$mdMedia('xs') }">
```

```js
angular.controller('ProductController', function( $scope, $products, $mdMedia) {
    $scope.$mdMedia           = $mdMedia;
}
```

**Pull request closed without merging**